Riot/Modules/Application/LegacyAppDelegate.m

```diff
@@ -1907,6 +1910,8 @@ - (void)initMatrixSessions
         {
             [self logoutWithConfirmation:NO completion:nil];
         }
+        
+        [self.delegate legacyAppDelegate:self didRemoveAccount:account];
```

Comment:
The `UserSessionsService.willRemoveUserSession` does not work well here. It will call after the logout.

What will be responsible for the logout? UserSessionsService?

Reply:
Yes at the end it should be `UserSessionsService` (or even a UserSessionService per user session if needed)